tests/Spec/Nri/Ui/Mark.elm

```diff
@@ -0,0 +1,377 @@
+module Spec.Nri.Ui.Mark exposing (..)
+
+import Accessibility.Styled.Style exposing (invisibleStyle)
+import Css
+import Expect
+import Markdown.Block
+import Markdown.Inline
+import Nri.Ui.Mark.V6 as Mark
+import String.Extra
+import Test exposing (..)
+
+
+startStyles : List Css.Style
+startStyles =
+    [ Css.color (Css.hex "000000") ]
+
+
+markStyles : List Css.Style
+markStyles =
+    [ Css.fontWeight Css.bold ]
+
+
+endStyles : List Css.Style
+endStyles =
+    [ Css.position Css.absolute ]
+
+
+start2Styles : List Css.Style
+start2Styles =
+    [ Css.color (Css.hex "222222") ]
+
+
+mark2Styles : List Css.Style
+mark2Styles =
+    [ Css.fontWeight Css.bolder ]
+
+
+end2Styles : List Css.Style
+end2Styles =
+    [ Css.position Css.fixed ]
+
+
+{-| Cloned from Mark for testing
+-}
+highlightDescription : String -> List Mark.Mark -> String
+highlightDescription prefix marks =
+    let
+        names =
+            String.Extra.toSentenceOxford (List.filterMap (.name >> Maybe.map stripMarkdownSyntax) marks)
+    in
+    if names == "" then
+        prefix ++ " highlight"
+
+    else if List.length marks == 1 then
+        prefix ++ " " ++ names ++ " highlight"
+
+    else
+        prefix ++ " " ++ names ++ " highlights"
+
+
+{-| Cloned from Mark for testing
+-}
+stripMarkdownSyntax : String -> String
+stripMarkdownSyntax markdown =
+    case Markdown.Block.parse Nothing markdown of
+        [ Markdown.Block.Paragraph _ inlines ] ->
+            Markdown.Inline.extractText inlines
+
+        _ ->
+            markdown
+
+
+{-| Cloned from Mark for testing
+-}
+cssContent : String -> Css.Style
+cssContent content =
+    Css.property "content" ("\" " ++ content ++ " \"")
+
+
+{-| Cloned from Mark for testing
+-}
+tagBeforeContent : List Mark.Mark -> List Css.Style
+tagBeforeContent marks =
+    if List.isEmpty marks then
+        []
+
+    else
+        [ Css.before
+            [ cssContent (highlightDescription "start" marks)
+            , invisibleStyle
+            ]
+        ]
+
+
+{-| Cloned from Mark for testing
+-}
+tagAfterContent : List Mark.Mark -> List Css.Style
+tagAfterContent marks =
+    if List.isEmpty marks then
+        []
+
+    else
+        [ Css.after
+            [ cssContent (highlightDescription "end" marks)
+            , invisibleStyle
+            ]
+        ]
+
+
+{-| Mark.overlappingStyles gives us a (Maybe (Html msg)) for the label outputs,
+but we can't test equality on HTML elements very well so we need to drop it.
```

Comment:
Maybe we could map the `Maybe` to a `Just MarkStartLabel | Nothing`?

There's one thing that's confusing to me about the tests below which is why `startStyles` are sometimes missing, and my guess is sometimes they're on this label, sometimes they're not.

Reply:
ps: I did a lazy attempt at this [here](https://github.com/NoRedInk/noredink-ui/pull/1669)

feel free to take it or leave it

Reply:
Ok, so this comment is old and needs to be updated - I had an incorrect understanding of what was happening before... 

If a mark has a name (which they really should for a11y purposes), then that is what gets put into the `Maybe (Html msg)` element as invisible content.  You can make the mark names show by calling `viewWithInlineTags` instead.  

So the start styles are **always** present, but if the mark has a name then they are applied to the invisible prefix element - if there is no name than the styles are attached to the first word itself. 

I'm going to see if I can make this clearer in the comments